### PR TITLE
Ignore '/docs' on some GH actions, give docs its own action

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

First time fiddling with GitHub Actions... be kind to me :)

Attempting to skip several checks for file changes within the `/docs` directory. Also adding a new action that lints/builds the docs, _only when_ files in `/docs` are changed.

### TEST PLAN

Gonna tweak a few files herein, and see how it responds.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: .github/workflows/prefer-typescript.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ name: Prefer TypeScript
 
 on:
   pull_request:
+    paths-ignore:
+      - 'docs/**'
```
Comment (ktmud) [inline .github/workflows/prefer-typescript.yml:6]: Don't we want to prefer TypeScript in docs?

File: docs/src/pages/404.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ const NotFoundPage = () => (
   <Layout>
     <SEO title="404: Not found" />
     <h1>NOT FOUND</h1>
-    <p>You just hit a route that does not exist... the sadness.</p>
```
Comment (ktmud) [inline docs/src/pages/404.jsx:28]: 😢

File: .github/workflows/superset-frontend.yml [hunk 3]
```diff
@@ -25,18 +31,3 @@ jobs:
       working-directory: ./superset-frontend
       run: |
         bash <(curl -s https://codecov.io/bash) -cF javascript
-
-  docs:
```
Comment (craig-rueda) [inline .github/workflows/superset-frontend.yml:29]: Was this even working? Looks like it was running from `./superset-frontend`.

**Review by craig-rueda — APPROVED** [review]
One comment. Agree that we want TS in docs :). Otherwise LGTM

File: .github/workflows/prefer-typescript.yml (hunk 1 above)
Comment (rusackas) [reply .github/workflows/prefer-typescript.yml:6]: Yes, but I figured we didn't want it muddying our metrics tracked with that action

File: .github/workflows/superset-frontend.yml (hunk 3 above)
Comment (rusackas) [reply .github/workflows/superset-frontend.yml:29]: Nope... but I hope it will now.

File: .github/workflows/prefer-typescript.yml (hunk 1 above)
Comment (rusackas) [reply .github/workflows/prefer-typescript.yml]: In any case, undid this one, thanks.

**Comment by nytai** [comment]
This change makes it so that PRs with only changes to `docs/**` can never pass CI since required checks won't run. See https://github.com/apache/incubator-superset/pull/11026. Can we configure this filter to pass instead of remaining in a pending state always?

**Comment by rusackas** [comment]
> This change makes it so that PRs with only changes to `docs/**` can never pass CI since required checks won't run. See #11026. Can we configure this filter to pass instead of remaining in a pending state forever?

@nytai @craig-rueda @ktmud 
Welp... from the docs, I don't see a way to effectively skip a test AND return a positive error code. I'm open to ideas if anybody has them, but my current inclination is to revert that change, and return to unnecessarily burning time/electricity.

**Comment by nytai** [comment]
@rusackas The only thing I can think of is to build the file/path checking step into the actual test code. I did something like this for the fossa check a while back, https://github.com/preset-io/incubator-superset/blob/master/.github/workflows/license-check.yml#L36

File: .github/workflows/superset-frontend.yml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 name: Frontend
 
-on: [push, pull_request]
+on:
+  push:
```
Comment (mistercrunch) [inline .github/workflows/superset-frontend.yml:4]: oh nice I didn't know you could do this! GH actions ftw!

Comment (ktmud) [reply .github/workflows/superset-frontend.yml:4]: My original GA implementation actually had some path ignores, but was [removed](https://github.com/apache/incubator-superset/pull/9538) out of abundance of caution.

Comment (nytai) [reply .github/workflows/superset-frontend.yml:4]: I think the main issue is that the check cannot be conditionally run _and_ required. We'd have to make all these checks not required and rely on reviewers/mergers to use their best judgement when merging.

**Pull request merged**